code/modules/admin/verbs/lua/lua_editor.dm

```diff
@@ -39,16 +42,19 @@
 /datum/lua_editor/ui_data(mob/user)
 	var/list/data = list()
 	data["noStateYet"] = !current_state
-	data["showDebugInfo"] = show_debug_info
+	data["showGlobalTable"] = show_global_table
 	if(current_state)
-		if(current_state.log && show_debug_info)
-			data["stateLog"] = kvpify_list(refify_list(current_state.log))
+		if(current_state.log)
+			data["stateLog"] = kvpify_list(refify_list(current_state.log.Copy((page*50)+1, min((page+1)*50+1, current_state.log.len+1))))
+		data["page"] = page
+		data["pageCount"] = CEILING(current_state.log.len/50, 1)
 		data["tasks"] = current_state.get_tasks()
-		if(show_debug_info)
+		if(show_global_table)
 			current_state.get_globals()
 			data["globals"] = kvpify_list(refify_list(current_state.globals))
 	data["states"] = SSlua.states
 	data["callArguments"] = kvpify_list(refify_list(arguments))
+	LAZYSET(tgui_shared_states, "shouldUpdateScroll", "true")
```

Comment:
You can not use this private tgui API. Expose the variable via ui_data instead if you need this much control.

Reply:
i was in the process of doing this as you were making this comment, so good to know I shouldn't use it.